py3status/modules/yubikey.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+"""
+Show an indicator when YubiKey is waiting for a touch.
+
+Configuration parameters:
+    format: Display format for the module.
+        (default '\?if=is_waiting YubiKey')
+    gpg_check_timeout: Time to wait for gpg check response.
+        Use value as small as possible that doesn't introduce false positives.
+        (default 0.1)
+    gpg_check_watch_paths: A list of files to watch, start gpg check if any one was opened.
+        (default ['~/.gnupg/pubring.kbx', '~/.ssh/known_hosts'])
+    sudo_check_watch_paths: A list of files to watch, toggle sudo check if any one was opened.
+        (default ['~/.config/Yubico/u2f_keys'])
+
+Control placeholders:
+    {is_waiting} a boolean indicating whether YubiKey is waiting for a touch.
+
+SAMPLE OUTPUT
+{'color': '#00FF00', 'full_text': 'YubiKey'}
+
+Dependencies:
+    gpg: to check for pending gpg access request
+    inotify-simple: to check for pending gpg and sudo requests
+    github.com/maximbaz/pam-u2f: a fork that adds watch capability to pam-u2f module
+    subprocess32: if after all these years you are still using python2
+
+@author Maxim Baz (https://github.com/maximbaz)
+@license BSD
+"""
+
+import time
+import os
+import signal
+import sys
+import threading
+import inotify_simple
+
+if os.name == 'posix' and sys.version_info[0] < 3:
+    import subprocess32 as subprocess
+else:
+    import subprocess
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    format = '\?if=is_waiting YubiKey'
+    gpg_check_timeout = 0.1
+    gpg_check_watch_paths = ['~/.gnupg/pubring.kbx', '~/.ssh/known_hosts']
+    sudo_check_watch_paths = ['~/.config/Yubico/u2f_keys']
+
+    def post_config_hook(self):
+        self.gpg_check_watch_paths = [
+            os.path.expanduser(p) for p in self.gpg_check_watch_paths
+        ]
+        self.sudo_check_watch_paths = [
+            os.path.expanduser(p) for p in self.sudo_check_watch_paths
+        ]
+
+        self.killed = threading.Event()
+
+        self.pending_gpg = False
+        self.pending_sudo = False
+
+        class GpgThread(threading.Thread):
```

Comment:
(Useful?) If you look at `volume_status`, you'll find several classes outside of `class Py3status`. If you are able to do the same here, then you could reclaim some indentations... and some things might be more readable... I recall we have to use `parent` for py3 stuffs... eg `self.parent.py3.update()`. Hopefully that's useful.

Something to try if we don't like how it's sitting inside `post_config_hook` instead of a line or two.

Reply:
It started as two very simple classes with just few lines, but since then they grew... Now it is time to extract them indeed, makes total sense to me.